libvast/src/logger.cpp

```diff
@@ -15,9 +15,250 @@
+
+} // namespace
+
+namespace detail {
+
+bool setup_spdlog(const vast::invocation& cmd_invocation,
+                  const caf::settings& cfg_file) {
+  // already running .....
+  if (vast_logger && vast_logger->name() != "/dev/null") {
+    VAST_ERROR_ANON("Log already up");
+    return false;
+  }
+
+  bool is_server = cmd_invocation.name() == "start"
+                   || caf::get_or(cmd_invocation.options, "vast.node", false);
+
+  const auto& cfg_cmd = cmd_invocation.options;
+
+  auto console_verbosity = vast::defaults::logger::console_verbosity ;
+  auto file_verbosity = vast::defaults::logger::file_verbosity ;
+
+  // The log level setup, just top down.
+  // Command line overwrites config file values
+  auto verbosity = caf::get_if<caf::atom_value>(&cfg_cmd, "vast.verbosity");
+  if (verbosity) {
+    if (loglevel_to_int(*verbosity, -1) < 0) {
+      std::cerr<< "Illegal vast.verbosity " << to_string(*verbosity) << "\n" ;
+      return false;
+    }
+    file_verbosity = *verbosity ;
+    console_verbosity = *verbosity;
+  } else {
+    auto cfg_console_verbosity = caf::get_if<std::string >(&cfg_file, "vast.console-verbosity");
+    if (cfg_console_verbosity) {
+      auto atom_cv = caf::atom_from_string(*cfg_console_verbosity) ;
+      if (loglevel_to_int(atom_cv, -1) < 0) {
+        std::cerr<< "Illegal vast.console-verbosity " << *cfg_console_verbosity << "\n" ;
+        return false;
+      } else {
+        console_verbosity = atom_cv ;
+      }
+    }
+    auto cfg_file_verbosity = caf::get_if<std::string >(&cfg_file, "vast.file-verbosity");
+    if (cfg_file_verbosity) {
+      auto atom_cv = caf::atom_from_string(*cfg_file_verbosity) ;
+      if (loglevel_to_int(atom_cv, -1) < 0) {
+        std::cerr<< "Illegal vast.file-verbosity " << *cfg_file_verbosity << "\n" ;
+        return false;
+      } else {
+        file_verbosity = atom_cv ;
+      }
+    }
+  }
+
+  auto vast_file_verbosity = loglevel_to_int(file_verbosity);
+  auto vast_console_verbosity = loglevel_to_int(console_verbosity);
+  auto vast_verbosity = std::max(vast_file_verbosity, vast_console_verbosity);
+
+  // Helper to set the color mode
+  spdlog::color_mode log_color = [&]() -> spdlog::color_mode {
+    auto config_value = caf::get_or(cfg_file, "vast.console", "automatic");
+    if (config_value == "automatic")
+      return spdlog::color_mode::automatic;
+    if (config_value == "always")
+      return spdlog::color_mode::always;
+
+    return spdlog::color_mode::never;
+  }();
+
+  auto log_file = caf::get_or(cfg_file, "vast.log-file",
+                              std::string{defaults::logger::log_file});
+
+  auto cmdline_log_file = caf::get_if<std::string>(&cfg_cmd, "vast.log-file");
+  if (cmdline_log_file) {
+    log_file = *cmdline_log_file;
+  }
+
+  if (is_server) {
+    path log_dir = caf::get_or(cfg_file, "vast.db-directory",
+                               defaults::system::db_directory);
+    if (!exists(log_dir)) {
```

Comment:
The original code has the additional conditions:

```
caf::get_or(cfg, "logger.file-name", default_fn) == default_fn
      && loglevel_to_int(file_verbosity) > VAST_LOG_LEVEL_QUIET
```

ie. we only attempt to create the log dir if the user didn't specify a custom log file (e.g. `/var/log/vast.log`)

Also, an important detail is that `log-file` is changed to `server.log` for this case in the original code. The `default_fn` is a CAF detail the we probably don't need anymore in the new code, so a simplified version could look like:

```
if (!exists(log_dir)
    && !get_if(&opts, "vast.log-file")
    &&  loglevel_to_int(file_verbosity) > VAST_LOG_LEVEL_QUIET) {
  log_file = (log_dir / "server.log").str();
  [...]
```

Reply:
Ad 1)
actually, I think, we do not need to create a directory at all, afaik spdlog will do it for us anyway, but we might lose the custom error message. That's why I let it as it is, but right , no logging no dir, so maybe forward the problem to spdlog than it solves itself and we do not need all the code.


Ad 2) 
indeed, I understood the yaml file wrong. 
Will update the logic to only use (and create) the db directory if the log file is the default one